setup.py

```diff
@@ -87,27 +94,45 @@ def get_version():
     return locals()['__version__']
 
 
-def make_cuda_ext(name, module, sources):
+def use_cuda():
+    return torch.cuda.is_available() or os.getenv('FORCE_CUDA', '0') == '1'
+
 
+def make_extension(name, module, sources, is_cuda_ext=False):
+    name = '{}.{}'.format(module, name)
+    sources = [os.path.join(*module.split('.'), p) for p in sources]
     define_macros = []
 
-    if torch.cuda.is_available() or os.getenv('FORCE_CUDA', '0') == '1':
-        define_macros += [('WITH_CUDA', None)]
+    is_cuda_ext = any([source.endswith('.cu') for source in sources])
```

Comment:
For operators end with `.cpp`, if compiled with cuda it will speed up a little bit.
It would be better if we could keep extensions unchanged when `CPU_ONLY==False`.